src/Parser.jl

```diff
@@ -381,9 +382,11 @@ function unparameterize_type(T::Type)
     candidate <: Union{} ? T : candidate
 end
 
-function parse(str::AbstractString; dicttype::Type{<:AbstractDict}=Dict{String,Any}, inttype::Type{<:Real}=Int64)
+function parse(str::AbstractString;
+               dicttype::Type{<:AbstractDict}=Dict{String,Any},
+               inttype::Type{<:Real}=Int64)
     pc = ParserContext{unparameterize_type(dicttype), inttype}()
-    ps = MemoryParserState(Vector{UInt8}(String(str)), 1)
+    ps = MemoryParserState(get_bytes(str), 1)
```

Comment:
It would be better not to make a copy here.

Since `MemoryParserState` is only ever created from a `String`, it would be better to just (a) change the `utf8data` field to a `String` field and (b) call `codeunit(s, i)` to extract a byte from the string (e.g. in the `byteat` function).  Note that `codeunit` already works in Julia 0.6.

Reply:
Yes, but I'm trying to make the minimal changes here (as I was told many times before to do 🙂 ) to make it work on both v0.6 and later.

Reply:
My suggestions require hardly any more code, and don't make more copies.

Reply:
They don't make less copies either.  My change actually requires gets rid of the copies on v0.7, and doesn't add them for v0.6.

Reply:
Your version has makes more copies on 0.7 than 0.6, whereas mine does not.

Reply:
On v0.7, it calls `codeunits`.  Isn't that what your's does on v0.7?